flink-container/src/test/java/org/apache/flink/container/entrypoint/ClassPathJobGraphRetrieverTest.java

```diff
@@ -160,6 +238,80 @@ public void testJarFromClassPathSupplier() throws IOException {
 		assertThat(jarFiles, contains(file1, file2));
 	}
 
+	@Test
+	public void testJobGraphRetrievalFailIfJobDirDoesNotHaveEntryClass() throws IOException {
+		final File testJar = TestJob.getTestJobJar();
+		final ClassPathJobGraphRetriever classPathJobGraphRetriever = new ClassPathJobGraphRetriever(
+			new JobID(),
+			SavepointRestoreSettings.none(),
+			PROGRAM_ARGUMENTS,
+			null,
+			// only find the entry class from job dir
+			() -> Collections.singleton(testJar),
+			userDirHasNotEntryClass);
```

Comment:
Should we maybe introduce a builder for the `ClassPathJobGraphRetriever` to give more expressive names and to better model which fields are optional (`jobClassName` and `jobDir`)?